core/shared/src/main/scala/scalaz/zio/Queue.scala

```diff
@@ -146,12 +137,9 @@ class Queue[A] private (
    * Future calls to `offer*` and `take*` will be interrupted immediately.
    */
   final val shutdown: UIO[Unit] = (for {
-    hook <- shutdownHook.modify {
-             case None       => (IO.unit, None)
-             case Some(hook) => (hook, None)
-           }
+    _      <- shutdownHook.succeed(())
```

Comment:
How about
```scala
shutdownHook.succeed(()) >>= 
  if (_)
    IO..effectTotal(unsafePollAll(takers)) >>= IO.foreachPar(_)(_.interrupt) *> strategy.shutdown
  else
    IO.unit
```

Reply:
Makes sense (I have to add some extra parentheses for Scala to infer everything properly)

Reply:
Even simpler with `IO.when`!

Reply:
Even simpler_er_ with `IO.whenM` 😆

Reply:
Oh right!